typing/typedecl.ml

```diff
@@ -499,6 +500,17 @@ let transl_labels (type rep) ~(record_form : rep record_form) ~new_var_jkind
          let modalities =
           Typemode.transl_modalities ~maturity:Stable mut modalities
          in
+         (if is_atomic
+          then
+            (* Forbid comonadic modalities on atomic fields, as we can't force
+               them to be respected when setting them via eg [%atomic.loc]. *)
+            match
+              Mode.Modality.Value.Const.equate modalities
```

Comment:
The error message calls out only future axes... but this looks like it checks past axes too. Is that intentional?

Reply:
I'm fine with that.